GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -195,6 +195,8 @@ private CancellationToken CancelBackgroundTasks()
 
             await this.SwitchToMainThreadAsync(token);
             _tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
+
+            await Task.WhenAll(_tasks).ConfigureAwait(false);
```

Comment:
The `_tasks` field needs to be removed (changed to local variable).

Reply:
Yes, it seems it was over-engineered.